assignments/this.js

```diff
@@ -1,26 +1,69 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. -Window/ global Object binding
+    - Here `this` refers to the window object, an object which has a global scope in a given context
+
+* 2. Implicit Binding
+    - When `this` is used inside an object, the binding is said to be implicit
+
+* 3. New Binding
+     - here `this` refers to the instance of the object returned by a constructor function
```

Comment:
'new' keyword makes a function call a constructor call. When a function is invoked with constructor call
a new object is created (constructed) and set as 'this' binding for that function call.